hack/embedded-repo.sh

```diff
@@ -0,0 +1,118 @@
+    "storage-ceph/${STORAGE_CEPH_IMAGE_ID}/installed-${STORAGE_CEPH_IMAGE_ID}-${NCN_ARCH}.repos"; do
+        curl -Ss -u "${ARTIFACTORY_USER}:${ARTIFACTORY_TOKEN}" "https://artifactory.algol60.net/artifactory/csm-images/stable/${REPOS_URL}"
+done | grep -E '^baseurl=https://' \
+     | sed -e 's/^baseurl=//' \
+     | sed -e 's|https://[^@]*@|https://|' \
+     | sed -e 's/\?auth=basic$//' \
+     | sort -u > "${TMPDIR}/ncn.repo-list.releasever"
+
+# Append update_debug repos, where kernel-default-debuginfo package is provided
+if [ -n "$KERNEL_DEFAULT_DEBUGINFO_VERSION" ]; then
+    echo 'https://artifactory.algol60.net/artifactory/sles-mirror/Updates/SLE-Module-Legacy/${releasever_major}-SP${releasever_minor}/${basearch}/update_debug' >> "${TMPDIR}/ncn.repo-list.releasever"
+fi
+
+# Try repos for SLES 15 SP3 and SP4
+(
+    cat "${TMPDIR}/ncn.repo-list.releasever" \
+        | sed -e "s/\${basearch}/${NCN_ARCH}/g" \
+        | sed -e "s/\${releasever_major}/15/g" \
+        | sed -e "s/\${releasever_minor}/4/g" \
+        | sed -e "s/\${releasever}/15.4/g"
+    cat "${TMPDIR}/ncn.repo-list.releasever" \
+        | sed -e "s/\${basearch}/${NCN_ARCH}/g" \
+        | sed -e "s/\${releasever_major}/15/g" \
+        | sed -e "s/\${releasever_minor}/3/g" \
+        | sed -e "s/\${releasever}/15.3/g"
+) | sort -u > "${TMPDIR}/ncn.repo-list.unverified"
+
+# Filter out non-existent repos and generate directory names for rpm-index input
+echo -ne > "${TMPDIR}/ncn.repo-list"
+cat "${TMPDIR}/ncn.repo-list.unverified" | while read url; do
+    if curl -I -Ss -f -u "${ARTIFACTORY_USER}:${ARTIFACTORY_TOKEN}" "$url/repodata/repomd.xml" >/dev/null 2>/dev/null; then
+        dir="${url#https://}"
+        dir="${dir#artifactory.algol60.net/artifactory/}"
+        dir="${dir//-mirror/}"
+        echo "$url" "$dir" >> "${TMPDIR}/ncn.repo-list"
+    fi
+done
+
+echo "List of repositories for embedded repo:"
+cat "${TMPDIR}/ncn.repo-list" | sed 's/^/    /'
+
+echo "Building RPM package index ..."
+export REPOCREDSVAR=$(jq --null-input --arg url "https://artifactory.algol60.net/artifactory/" --arg realm "Artifactory Realm" --arg user "$ARTIFACTORY_USER" --arg password "$ARTIFACTORY_TOKEN" '{($url): {"realm": $realm, "user": $user, "password": $password}}')
+# Filtering out conntrack package, because it is not in our repos, and gets into NCN image from local mount during build
+(cat "${TMPDIR}/ncn.rpm-list" \
+    | grep -v conntrack-1-1 \
+    | docker run -e REPOCREDSVAR --rm -i "${PACKAGING_TOOLS_IMAGE}" rpm-index -c REPOCREDSVAR -v \
+    --input-format NEVR \
+    --output-format DOWNLOAD_CSV \
+    $(sed -e 's/^/-d /' "${TMPDIR}/ncn.repo-list") \
+   -
+)> "${TMPDIR}/embedded.url-list"
+
+if [ "${1}" == "--validate" ]; then
+    echo "All RPM packages were resolved successfully"
+else
+    TARGET_DIR=$(realpath "${1}")
+    cat "${TMPDIR}/embedded.url-list" | while IFS="," read -r dir url; do
```

Comment:
Download files one by one with `curl`, instead of running `rpm-sync`. Somehow `rpm-sync` consumes 100% of CPU power and lasts ~ 3 times longer, even if instructed to use just a single thread. I suspect it may be due to known issue with Python running on Alpine docker container (https://pythonspeed.com/articles/alpine-docker-python/).